README.md

```diff
@@ -8,7 +8,130 @@ The plugin template is meant to be used as a starting point for OBS Studio plugi
+[^2]: Only if the plugin provides its own Qt-based widgets or dialogs
+[^3]: Only if the plugin interacts with the OBS Studio frontend via its frontend API
+[^4]: Can be enabled to speed up local builds if necessary, refer to Ccache documentation about possible caveats
+[^5]: By default `RelWithDebInfo` is used, CI will also use `Release` to generate release builds.
+[^6]: This should match the same config used to build the project in the prior step.
+
+### Debug Steps
+
+Debugging your plugin requires an existing build of OBS-Studio, preferably a self-compiled build using Debug config. Please follow the instructrions on the `obs-studio` repository to build your own local version of the application.
+
+#### Windows
+
+The build system will automatically copy the plugin as well as its resources into a configuration-specific directory called `rundir` found in the project build directory. A Debug build of the plugin can thus be found in `<Path-To-Checkout>/build_<ARCHITECTURE>/rundir/Debug`. The plugin consists of its binary file (a `.dll` library) and its resources subdirectory (which should have the same name as the `.dll`).
+
+To debug the code via MSVC, a few setup steps are necessary:
+
+1. Change the target used for launching the project from "ALL_BUILD" (CMake's default) to your plugin target.
+2. Change the "Debugger" configuration of your plugin target:
+    * Change the "Command" by browsing for the OBS Studio executable `obs64.exe` (preferrably a variant compiled locally)
+    * Change the "Working Directory" by browsing for the directory of that same `obs64.exe` was found in
+    * Change the "Environment" to contain two additional variables:
+        * `OBS_PLUGINS_PATH`
+        * `OBS_PLUGINS_DATA_PATH`
+        * Use the rundir location for both variables `<Path-To-Checkout>/build_<ARCHITECTURE>/rundir/Debug`
+3. Launch a debugging session just like with any other project
+
+#### macOS
+
+The build system will automatically copy the plugin bundle into a configuration-specific directory called `rundir` found in the project build directory. A Debug build of the plugin can thus be found in `<Path-To-Checkout>/build_macos/rundir/Debug`.
+
+To debug the code via Xcode, a few setup steps are necessary:
+
+1. Copy the plugin bundle from the rundir to the OBS Studio plugin directory found in `/Users/<Your Username>/Library/Application Support/obs-studio/plugins`
+2. Click on the current scheme used by Xcode and then click "Edit Scheme"
+3. Select your plugin target from the list of schemes
+4. In the "Run" configuration and under "Info", change the "Executable" by browsing for your local build of `OBS.app` (debugging won't work with official releases)
+5. Under "Arguments", set up two environment variables:
+    * `OBS_PLUGINS_PATH`
+    * `OBS_PLUGINS_DATA_PATH`
+    * Use the rundir location for both variables `<Path-To-Checkout>/build_<ARCHITECTURE>/rundir/Debug`
```

Comment:
Aren't steps 1 and 5 redundant? If you copy the bundle to the plugin directory, there's no need to set the paths, or when you set the paths there should be no need to copy the bundle.

Reply:
That depends on when https://github.com/obsproject/obs-studio/pull/11468 is merged. I'd expect this PR to be in flux until that happens.

`build_macos/Debug` is managed by the build system - we have no control over that one and rather not rely on it.

Reply:
I see.

Reply:
That PR has been merged.